packages/datagateway-download/src/downloadApiHooks.ts

```diff
@@ -121,43 +179,62 @@ export const useIsTwoLevel = (): UseQueryResult<boolean, AxiosError> => {
   });
 };
 
-// TODO: refactor rest of dg-download to use react-query
-// export const useSubmitCart = (): UseMutationResult<
-//   number,
-//   AxiosError,
-//   {
-//     transport: string;
-//     emailAddress: string;
-//     fileName: string;
-//     zipType?: 'ZIP' | 'ZIP_AND_COMPRESS';
-//   }
-// > => {
-//   const queryClient = useQueryClient();
-//   const settings = React.useContext(DownloadSettingsContext);
-//   const { facilityName, downloadApiUrl } = settings;
-
-//   return useMutation(
-//     ({ transport, emailAddress, fileName, zipType }) =>
-//       submitCart(
-//         transport,
-//         emailAddress,
-//         fileName,
-//         {
-//           facilityName,
-//           downloadApiUrl,
-//         },
-//         zipType
-//       ),
-//     {
-//       onSuccess: (data) => {
-//         queryClient.setQueryData('cart', data);
-//       },
-//       onError: (error) => {
-//         handleICATError(error);
-//       },
-//     }
-//   );
-// };
+export interface SubmitCartParams {
+  transport: string;
+  emailAddress: string;
+  fileName: string;
+  zipType?: SubmitCartZipType;
+}
+
+/**
+ * A React hook for submitting a download cart.
+ * Returns the download id for the submitted cart, which can then be used
+ * to query more info.
+ */
+export const useSubmitCart = (
+  options?: UseMutationOptions<
+    number,
+    AxiosError,
+    SubmitCartParams,
+    RollbackFunction
+  >
+): UseMutationResult<
+  number,
+  AxiosError,
+  SubmitCartParams,
+  RollbackFunction
+> => {
+  const queryClient = useQueryClient();
+  const settings = React.useContext(DownloadSettingsContext);
+  const { facilityName, downloadApiUrl } = settings;
+
+  return useMutation(
+    ({ transport, emailAddress, fileName, zipType }) =>
+      submitCart(
+        transport,
+        emailAddress,
+        fileName,
+        {
+          facilityName,
+          downloadApiUrl,
+        },
+        zipType
+      ),
+    {
+      onError: (error, _, rollback) => {
+        console.log('error', error);
```

Comment:
We don't need to log the error here - `handleICATError` already logs the error message